standalone-metastore/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java

```diff
@@ -868,6 +875,204 @@ public void commitTxn(CommitTxnRequest rqst)
     }
   }
 
+  @Override
+  @RetrySemantics.ReadOnly
+  public GetOpenWriteIdsResponse getOpenWriteIds(GetOpenWriteIdsRequest rqst)
+          throws NoSuchTxnException, MetaException {
+    try {
+      // We need to figure out the current transaction number and the list of
+      // open transactions.  To avoid needing a transaction on the underlying
+      // database we'll look at the current transaction number first.  If it
+      // subsequently shows up in the open list that's ok.
+      Connection dbConn = null;
+      Statement stmt = null;
+      ResultSet rs = null;
+      ValidTxnList validTxnList;
+      if (rqst.isSetValidTxnStr()) {
+        validTxnList = new ValidReadTxnList(rqst.getValidTxnStr());
+      } else {
+        validTxnList = TxnUtils.createValidReadTxnList(getOpenTxns(), 0);
```

Comment:
If the caller such as compactor isn't associated with any txn , then will expect to get the current snapshot instead of depending on any txn snapshot. In such cases, they would pass null for validTxnStr.